### ISSUE TITLE
Internal error formatting py2 code

### ISSUE THREAD

**Issue by theunkn0wn1** [psf/black#2107]
**Describe the bug** Attempted to reformat python2.7 (i presume its 2.7...) code, reslted in an INTERNAL ERROR

**To Reproduce** Steps to reproduce the behavior:

1. gh repo clone krejov100/EtherSense
2. cd EtherSense
3. black -t py27 .

**Expected behavior** 
all python modules to reformat successfully

**Environment (please complete the following information):**
```
Linux crypt 5.10.17-v7l+ #1403 SMP Mon Feb 22 11:33:35 GMT 2021 armv7l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Wed Apr 14 19:45:09 2021 from 192.168.7.117
...

Python 3.7.3 (default, Jan 22 2021, 20:04:44) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.22.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import black

In [2]: black.__version__
Out[2]: '20.8b1'
``

**Does this bug also happen on master?** To answer this, you have two options:

1. Use the online formatter at https://black.now.sh/?version=master, which will use the
   latest master branch.
2. Or run _Black_ on your machine:
   - create a new virtualenv (make sure it's the same Python version);
   - clone this repository;
   - run `pip install -e .[d,python2]`;
   - run `pip install -r test_requirements.txt`
   - make sure it's sane by running `python -m unittest`; and
   - run `black` like you did last time.
Used route 2, tests passed
```
----------------------------------------------------------------------
Ran 179 tests in 133.400s

OK (expected failures=3)

```

Issue persists
**Additional context** Add any other context about the problem here.

It occurs to me this is in a py2x codebase So i am not expecting that much support; further im leaning towards a total rewrite of it since its not a py3 compatible codebase. Only reporting this here since the `black` tool asked me to.

**Comment by ichard26** [comment]
Hi,

The problem seems to be that your `EtherSenseServer.py` file has mixed tab/space indentation. 

![Screenshot from 2021-04-14 18-38-45](https://user-images.githubusercontent.com/63936253/114789360-019e5e80-9d51-11eb-80cf-ddfe8f4708b2.png)

This form of indentation is hard to parse correctly since it's unclear what amount of spaces counts a single tab. Matter of fact, this file is rejected by Python 3 for this exact issue:

```console
ichard26@acer-ubuntu:~/programming/oss/EtherSense$ python EtherSenseServer.py 
  File "EtherSenseServer.py", line 25
    depth2 = depth_filter.process(depth)
                                       ^
TabError: inconsistent use of tabs and spaces in indentation
```

Black does has some indentation fixing functionalities but they aren't designed for a case like this. Attempting to fix mixed indentation is inherently unsafe and I'm surprised Black gets it right (I think) reviewing its output when disabling the safety checks (--fast). I recommend fixing the indentation to be at least consistent (although Black will format it to 4 spaces) and then trying Black again.

> It occurs to me this is in a py2x codebase So i am not expecting that much support

Black does explicitly support formatting Python 2 code (afterall `-t py27` is a valid option) so technically you should expect the same level of support, although in practice it's hard to provide the same level of support. For instance, I actually haven't used Python 2 before, so debugging Python 2 specific issues is a lot more difficult for me.

I hope that helps!

**Comment by theunkn0wn1** [comment]
Thanks for the response!

> The problem seems to be that your EtherSenseServer.py file has mixed tab/space indentation.

That is exactly the case, and was actually the reason I attempted to feed the file to black in the hopes it could handle fixing that for me. If it can't, its not too troubling in this case; I am already considering a total rewrite (which would fix other issues in the codebase)

> I actually haven't used Python 2 before, so debugging Python 2 specific issues is a lot more difficult for me.
It gets more "interesting" when you have to debug python2->3 migration issues 🙃

**Comment by ichard26** [comment]
I mean, you can try `black EtherSenseServer.py --fast -t py27` which disables the safety checks, a cursory look tells me that for your case Black does fix the indentation correctly. No promises this will work for all cases though.

**Comment by kamahen** [comment]
If you have tabs in your code, and black is the wrong tool for fixing them, you might use something like Emacs' `untabify`, with various options about where and how tabs get expanded (vim has "retab" apparently).

**Comment by ichard26** [comment]
I'm going to close this issue as a duplicate of GH-952 since ultimately the issue here is that Black doesn't handle mixed tab/space indentation in a graceful manner. Why now? I randomly stumbled upon that issue just now, it's hard to remember what issues you've seen before!
